### DISCUSSION TITLE
posts/web-virtual-principle

### DISCUSSION THREAD

**Discussion by giscus** [hehehai/h-blog#2] [Announcements]
# posts/web-virtual-principle

网页中大量数据的列表渲染总是会出现性能问题，我们使用虚拟的空间来表示列表中不可见的元素，从而高度还原列表的展示和交互。

https://hehehai.cn/posts/web-virtual-principle

**Comment by xiaosongmao123** [comment]
InnerList使用前后Padding ，印象里虚拟列表使用transform性能更高，但不知道实际表现到底有多少差距，请试一下
参见 
https://ant-design.gitee.io/components/list-cn/#API
https://github.com/react-component/virtual-list

**Reply by hehehai** [reply]
嗯，是的 transform 理论上渲染性能是优于 padding 的。
